Deprecation warnings with GTK 3.20.x

``` sh
➤ go get github.com/gotk3/gotk3/gtk
# github.com/gotk3/gotk3/gdk
code/go/src/github.com/gotk3/gotk3/gdk/gdk.go:288:6: warning: 'gdk_device_grab' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gdk/gdkdevice.h:250:15: note: 'gdk_device_grab' has been explicitly marked deprecated here
code/go/src/github.com/gotk3/gotk3/gdk/gdk.go:305:25: warning: 'gdk_device_manager_get_client_pointer' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gdk/gdkdevicemanager.h:44:14: note: 'gdk_device_manager_get_client_pointer' has been explicitly marked deprecated here
code/go/src/github.com/gotk3/gotk3/gdk/gdk.go:357:25: warning: 'gdk_device_manager_list_devices' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gdk/gdkdevicemanager.h:41:14: note: 'gdk_device_manager_list_devices' has been explicitly marked deprecated here
code/go/src/github.com/gotk3/gotk3/gdk/gdk.go:373:2: warning: 'gdk_device_ungrab' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gdk/gdkdevice.h:259:15: note: 'gdk_device_ungrab' has been explicitly marked deprecated here
code/go/src/github.com/gotk3/gotk3/gdk/gdk.go:511:25: warning: 'gdk_display_get_device_manager' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gdk/gdkdisplay.h:170:20: note: 'gdk_display_get_device_manager' has been explicitly marked deprecated here
# github.com/gotk3/gotk3/gtk
code/go/src/github.com/gotk3/gotk3/gtk/gtk.go:1417:6: warning: 'gtk_button_get_focus_on_click' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gtk/gtkbutton.h:141:23: note: 'gtk_button_get_focus_on_click' has been explicitly marked deprecated here
code/go/src/github.com/gotk3/gotk3/gtk/gtk.go:1601:2: warning: 'gtk_button_set_focus_on_click' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gtk/gtkbutton.h:138:23: note: 'gtk_button_set_focus_on_click' has been explicitly marked deprecated here
# github.com/gotk3/gotk3/gtk
code/go/src/github.com/gotk3/gotk3/gtk/text_iter.go:260:6: warning: 'gtk_text_iter_begins_tag' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gtk/gtktextiter.h:178:10: note: 'gtk_text_iter_begins_tag' has been explicitly marked deprecated here
# github.com/gotk3/gotk3/gtk
code/go/src/github.com/gotk3/gotk3/gtk/window.go:703:2: warning: 'gtk_window_resize_to_geometry' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gtk/gtkwindow.h:451:6: note: 'gtk_window_resize_to_geometry' has been explicitly marked deprecated here
code/go/src/github.com/gotk3/gotk3/gtk/window.go:771:2: warning: 'gtk_window_set_default_geometry' is deprecated [-Wdeprecated-declarations]
/usr/include/gtk-3.0/gtk/gtkwindow.h:447:6: note: 'gtk_window_set_default_geometry' has been explicitly marked deprecated here
```

I'm busy atm, so if someone gets around fixing it, please leave a note, otherwise I'll eventually submit a PR.

--- a/gdk/gdk.go
+++ b/gdk/gdk.go
@@ -352,25 +352,6 @@ func marshalDevice(p uintptr) (interface{}, error) {
 	return &Device{obj}, nil
 }
 
-// Grab() is a wrapper around gdk_device_grab().
-func (v *Device) Grab(w *Window, ownership GrabOwnership, owner_events bool, event_mask EventMask, cursor *Cursor, time uint32) GrabStatus {
-	ret := C.gdk_device_grab(
-		v.native(),
-		w.native(),
-		C.GdkGrabOwnership(ownership),
-		gbool(owner_events),
-		C.GdkEventMask(event_mask),
-		cursor.native(),
-		C.guint32(time),
-	)
-	return GrabStatus(ret)
-}
-
-// Ungrab() is a wrapper around gdk_device_ungrab().
-func (v *Device) Ungrab(time uint32) {
-	C.gdk_device_ungrab(v.native(), C.guint32(time))
-}
-
 /*
  * GdkCursor
  */
@@ -444,18 +425,6 @@ func marshalDeviceManager(p uintptr) (interface{}, error) {
 	return &DeviceManager{obj}, nil
 }
 
-// GetClientPointer() is a wrapper around gdk_device_manager_get_client_pointer().
-func (v *DeviceManager) GetClientPointer() (*Device, error) {
-	c := C.gdk_device_manager_get_client_pointer(v.native())
-	if c == nil {
-		return nil, nilPtrErr
-	}
-	obj := &glib.Object{glib.ToGObject(unsafe.Pointer(c))}
-	obj.Ref()
-	runtime.SetFinalizer(obj, (*glib.Object).Unref)
-	return &Device{obj}, nil
-}
-
 // GetDisplay() is a wrapper around gdk_device_manager_get_display().
 func (v *DeviceManager) GetDisplay() (*Display, error) {
 	c := C.gdk_device_manager_get_display(v.native())
@@ -466,22 +435,6 @@ func (v *DeviceManager) GetDisplay() (*Display, error) {
 	obj.Ref()
 	runtime.SetFinalizer(obj, (*glib.Object).Unref)
 	return &Display{obj}, nil
-}
-
-// ListDevices() is a wrapper around gdk_device_manager_list_devices().
-func (v *DeviceManager) ListDevices(tp DeviceType) *glib.List {
-	clist := C.gdk_device_manager_list_devices(v.native(), C.GdkDeviceType(tp))
-	if clist == nil {
-		return nil
-	}
-	glist := glib.WrapList(uintptr(unsafe.Pointer(clist)))
-	glist.DataWrapper(func(ptr unsafe.Pointer) interface{} {
-		return &Device{&glib.Object{glib.ToGObject(ptr)}}
-	})
-	runtime.SetFinalizer(glist, func(glist *glib.List) {
-		glist.Free()
-	})
-	return glist
 }
 
 /*
@@ -558,19 +511,6 @@ func (v *Display) GetName() (string, error) {
 	return C.GoString((*C.char)(c)), nil
 }
 
-// GetScreen() is a wrapper around gdk_display_get_screen().
-func (v *Display) GetScreen(screenNum int) (*Screen, error) {
-	c := C.gdk_display_get_screen(v.native(), C.gint(screenNum))
-	if c == nil {
-		return nil, nilPtrErr
-	}
-	obj := &glib.Object{glib.ToGObject(unsafe.Pointer(c))}
-	s := &Screen{obj}
-	obj.Ref()
-	runtime.SetFinalizer(obj, (*glib.Object).Unref)
-	return s, nil
-}
-
 // GetDefaultScreen() is a wrapper around gdk_display_get_default_screen().
 func (v *Display) GetDefaultScreen() (*Screen, error) {
 	c := C.gdk_display_get_default_screen(v.native())
@@ -582,19 +522,6 @@ func (v *Display) GetDefaultScreen() (*Screen, error) {
 	obj.Ref()
 	runtime.SetFinalizer(obj, (*glib.Object).Unref)
 	return s, nil
-}
-
-// GetDeviceManager() is a wrapper around gdk_display_get_device_manager().
-func (v *Display) GetDeviceManager() (*DeviceManager, error) {
-	c := C.gdk_display_get_device_manager(v.native())
-	if c == nil {
-		return nil, nilPtrErr
-	}
-	obj := &glib.Object{glib.ToGObject(unsafe.Pointer(c))}
-	d := &DeviceManager{obj}
-	obj.Ref()
-	runtime.SetFinalizer(obj, (*glib.Object).Unref)
-	return d, nil
 }
 
 // DeviceIsGrabbed() is a wrapper around gdk_display_device_is_grabbed().

--- a/gdk/gdk_deprecated_since_3_20.go
+++ b/gdk/gdk_deprecated_since_3_20.go
@@ -1,0 +1,86 @@
+//+build gtk_3_6 gtk_3_8 gtk_3_10 gtk_3_12 gtk_3_14 gtk_3_16 gtk_3_18
+
+package gdk
+
+// #cgo pkg-config: gdk-3.0
+// #include <gdk/gdk.h>
+import "C"
+import (
+	"runtime"
+	"unsafe"
+
+	"github.com/gotk3/gotk3/glib"
+)
+
+// Grab() is a wrapper around gdk_device_grab().
+func (v *Device) Grab(w *Window, ownership GrabOwnership, owner_events bool, event_mask EventMask, cursor *Cursor, time uint32) GrabStatus {
+	ret := C.gdk_device_grab(
+		v.native(),
+		w.native(),
+		C.GdkGrabOwnership(ownership),
+		gbool(owner_events),
+		C.GdkEventMask(event_mask),
+		cursor.native(),
+		C.guint32(time),
+	)
+	return GrabStatus(ret)
+}
+
+// GetClientPointer() is a wrapper around gdk_device_manager_get_client_pointer().
+func (v *DeviceManager) GetClientPointer() (*Device, error) {
+	c := C.gdk_device_manager_get_client_pointer(v.native())
+	if c == nil {
+		return nil, nilPtrErr
+	}
+	obj := &glib.Object{glib.ToGObject(unsafe.Pointer(c))}
+	obj.Ref()
+	runtime.SetFinalizer(obj, (*glib.Object).Unref)
+	return &Device{obj}, nil
+}
+
+// ListDevices() is a wrapper around gdk_device_manager_list_devices().
+func (v *DeviceManager) ListDevices(tp DeviceType) *glib.List {
+	clist := C.gdk_device_manager_list_devices(v.native(), C.GdkDeviceType(tp))
+	if clist == nil {
+		return nil
+	}
+	glist := glib.WrapList(uintptr(unsafe.Pointer(clist)))
+	glist.DataWrapper(func(ptr unsafe.Pointer) interface{} {
+		return &Device{&glib.Object{glib.ToGObject(ptr)}}
+	})
+	runtime.SetFinalizer(glist, func(glist *glib.List) {
+		glist.Free()
+	})
+	return glist
+}
+
+// Ungrab() is a wrapper around gdk_device_ungrab().
+func (v *Device) Ungrab(time uint32) {
+	C.gdk_device_ungrab(v.native(), C.guint32(time))
+}
+
+// GetDeviceManager() is a wrapper around gdk_display_get_device_manager().
+func (v *Display) GetDeviceManager() (*DeviceManager, error) {
+	c := C.gdk_display_get_device_manager(v.native())
+	if c == nil {
+		return nil, nilPtrErr
+	}
+	obj := &glib.Object{glib.ToGObject(unsafe.Pointer(c))}
+	d := &DeviceManager{obj}
+	obj.Ref()
+	runtime.SetFinalizer(obj, (*glib.Object).Unref)
+	return d, nil
+}
+
+// GetScreen() is a wrapper around gdk_display_get_screen().
+func (v *Display) GetScreen(screenNum int) (*Screen, error) {
+	c := C.gdk_display_get_screen(v.native(), C.gint(screenNum))
+	if c == nil {
+		return nil, nilPtrErr
+	}
+	obj := &glib.Object{glib.ToGObject(unsafe.Pointer(c))}
+	s := &Screen{obj}
+	obj.Ref()
+	runtime.SetFinalizer(obj, (*glib.Object).Unref)
+	return s, nil
+}

--- a/gdk/gdk_deprecated_since_3_22.go
+++ b/gdk/gdk_deprecated_since_3_22.go
@@ -1,0 +1,84 @@
+//+build gtk_3_6 gtk_3_8 gtk_3_10 gtk_3_12 gtk_3_14 gtk_3_16 gtk_3_18 gtk_3_20
+
+package gdk
+
+// #cgo pkg-config: gdk-3.0
+// #include <gdk/gdk.h>
+import "C"
+
+// GetActiveWindow is a wrapper around gdk_screen_get_active_window().
+func (v *Screen) GetActiveWindow() (*Window, error) {
+	return toWindow(C.gdk_screen_get_active_window(v.native()))
+}
+
+// GetHeight is a wrapper around gdk_screen_get_height().
+func (v *Screen) GetHeight() int {
+	c := C.gdk_screen_get_height(v.native())
+	return int(c)
+}
+
+// GetHeightMM is a wrapper around gdk_screen_get_height_mm().
+func (v *Screen) GetHeightMM() int {
+	return int(C.gdk_screen_get_height_mm(v.native()))
+}
+
+// GetMonitorAtPoint is a wrapper around gdk_screen_get_monitor_at_point().
+func (v *Screen) GetMonitorAtPoint(x, y int) int {
+	return int(C.gdk_screen_get_monitor_at_point(v.native(), C.gint(x), C.gint(y)))
+}
+
+// GetMonitorAtWindow is a wrapper around gdk_screen_get_monitor_at_window().
+func (v *Screen) GetMonitorAtWindow(w *Window) int {
+	return int(C.gdk_screen_get_monitor_at_window(v.native(), w.native()))
+}
+
+// GetMonitorHeightMM is a wrapper around gdk_screen_get_monitor_height_mm().
+func (v *Screen) GetMonitorHeightMM(m int) int {
+	return int(C.gdk_screen_get_monitor_height_mm(v.native(), C.gint(m)))
+}
+
+// GetMonitorPlugName is a wrapper around gdk_screen_get_monitor_plug_name().
+func (v *Screen) GetMonitorPlugName(m int) (string, error) {
+	return toString(C.gdk_screen_get_monitor_plug_name(v.native(), C.gint(m)))
+}
+
+// GetMonitorScaleFactor is a wrapper around gdk_screen_get_monitor_scale_factor().
+func (v *Screen) GetMonitorScaleFactor(m int) int {
+	return int(C.gdk_screen_get_monitor_scale_factor(v.native(), C.gint(m)))
+}
+
+// GetMonitorWidthMM is a wrapper around gdk_screen_get_monitor_width_mm().
+func (v *Screen) GetMonitorWidthMM(m int) int {
+	return int(C.gdk_screen_get_monitor_width_mm(v.native(), C.gint(m)))
+}
+
+// GetNMonitors is a wrapper around gdk_screen_get_n_monitors().
+func (v *Screen) GetNMonitors() int {
+	return int(C.gdk_screen_get_n_monitors(v.native()))
+}
+
+// GetNumber is a wrapper around gdk_screen_get_number().
+func (v *Screen) GetNumber() int {
+	return int(C.gdk_screen_get_number(v.native()))
+}
+
+// GetPrimaryMonitor is a wrapper around gdk_screen_get_primary_monitor().
+func (v *Screen) GetPrimaryMonitor() int {
+	return int(C.gdk_screen_get_primary_monitor(v.native()))
+}
+
+// GetWidth is a wrapper around gdk_screen_get_width().
+func (v *Screen) GetWidth() int {
+	c := C.gdk_screen_get_width(v.native())
+	return int(c)
+}
+
+// GetWidthMM is a wrapper around gdk_screen_get_width_mm().
+func (v *Screen) GetWidthMM() int {
+	return int(C.gdk_screen_get_width_mm(v.native()))
+}
+
+// MakeDisplayName is a wrapper around gdk_screen_make_display_name().
+func (v *Screen) MakeDisplayName() (string, error) {
+	return toString(C.gdk_screen_make_display_name(v.native()))
+}

--- a/gdk/screen.go
+++ b/gdk/screen.go
@@ -74,18 +74,6 @@ func (v *Screen) GetSystemVisual() (*Visual, error) {
 	return visual, nil
 }
 
-// GetWidth is a wrapper around gdk_screen_get_width().
-func (v *Screen) GetWidth() int {
-	c := C.gdk_screen_get_width(v.native())
-	return int(c)
-}
-
-// GetHeight is a wrapper around gdk_screen_get_height().
-func (v *Screen) GetHeight() int {
-	c := C.gdk_screen_get_height(v.native())
-	return int(c)
-}
-
 // ScreenGetDefault is a wrapper aroud gdk_screen_get_default().
 func ScreenGetDefault() (*Screen, error) {
 	return toScreen(C.gdk_screen_get_default())
@@ -106,71 +94,11 @@ func (v *Screen) GetDisplay() (*Display, error) {
 	return toDisplay(C.gdk_screen_get_display(v.native()))
 }
 
-// GetNumber is a wrapper around gdk_screen_get_number().
-func (v *Screen) GetNumber() int {
-	return int(C.gdk_screen_get_number(v.native()))
-}
-
-// GetWidthMM is a wrapper around gdk_screen_get_width_mm().
-func (v *Screen) GetWidthMM() int {
-	return int(C.gdk_screen_get_width_mm(v.native()))
-}
-
-// GetHeightMM is a wrapper around gdk_screen_get_height_mm().
-func (v *Screen) GetHeightMM() int {
-	return int(C.gdk_screen_get_height_mm(v.native()))
-}
-
 func toString(c *C.gchar) (string, error) {
 	if c == nil {
 		return "", nilPtrErr
 	}
 	return C.GoString((*C.char)(c)), nil
-}
-
-// MakeDisplayName is a wrapper around gdk_screen_make_display_name().
-func (v *Screen) MakeDisplayName() (string, error) {
-	return toString(C.gdk_screen_make_display_name(v.native()))
-}
-
-// GetNMonitors is a wrapper around gdk_screen_get_n_monitors().
-func (v *Screen) GetNMonitors() int {
-	return int(C.gdk_screen_get_n_monitors(v.native()))
-}
-
-// GetPrimaryMonitor is a wrapper around gdk_screen_get_primary_monitor().
-func (v *Screen) GetPrimaryMonitor() int {
-	return int(C.gdk_screen_get_primary_monitor(v.native()))
-}
-
-// GetMonitorAtPoint is a wrapper around gdk_screen_get_monitor_at_point().
-func (v *Screen) GetMonitorAtPoint(x, y int) int {
-	return int(C.gdk_screen_get_monitor_at_point(v.native(), C.gint(x), C.gint(y)))
-}
-
-// GetMonitorAtWindow is a wrapper around gdk_screen_get_monitor_at_window().
-func (v *Screen) GetMonitorAtWindow(w *Window) int {
-	return int(C.gdk_screen_get_monitor_at_window(v.native(), w.native()))
-}
-
-// GetMonitorHeightMM is a wrapper around gdk_screen_get_monitor_height_mm().
-func (v *Screen) GetMonitorHeightMM(m int) int {
-	return int(C.gdk_screen_get_monitor_height_mm(v.native(), C.gint(m)))
-}
-
-// GetMonitorWidthMM is a wrapper around gdk_screen_get_monitor_width_mm().
-func (v *Screen) GetMonitorWidthMM(m int) int {
-	return int(C.gdk_screen_get_monitor_width_mm(v.native(), C.gint(m)))
-}
-
-// GetMonitorPlugName is a wrapper around gdk_screen_get_monitor_plug_name().
-func (v *Screen) GetMonitorPlugName(m int) (string, error) {
-	return toString(C.gdk_screen_get_monitor_plug_name(v.native(), C.gint(m)))
-}
-
-// GetMonitorScaleFactor is a wrapper around gdk_screen_get_monitor_scale_factor().
-func (v *Screen) GetMonitorScaleFactor(m int) int {
-	return int(C.gdk_screen_get_monitor_scale_factor(v.native(), C.gint(m)))
 }
 
 // GetResolution is a wrapper around gdk_screen_get_resolution().
@@ -181,11 +109,6 @@ func (v *Screen) GetResolution() float64 {
 // SetResolution is a wrapper around gdk_screen_set_resolution().
 func (v *Screen) SetResolution(r float64) {
 	C.gdk_screen_set_resolution(v.native(), C.gdouble(r))
-}
-
-// GetActiveWindow is a wrapper around gdk_screen_get_active_window().
-func (v *Screen) GetActiveWindow() (*Window, error) {
-	return toWindow(C.gdk_screen_get_active_window(v.native()))
 }
 
 // void 	gdk_screen_set_font_options ()

--- a/gtk/gtk.go
+++ b/gtk/gtk.go
@@ -1461,17 +1461,6 @@ func (v *Button) GetUseUnderline() bool {
 	return gobool(c)
 }
 
-// SetFocusOnClick() is a wrapper around gtk_button_set_focus_on_click().
-func (v *Button) SetFocusOnClick(focusOnClick bool) {
-	C.gtk_button_set_focus_on_click(v.native(), gbool(focusOnClick))
-}
-
-// GetFocusOnClick() is a wrapper around gtk_button_get_focus_on_click().
-func (v *Button) GetFocusOnClick() bool {
-	c := C.gtk_button_get_focus_on_click(v.native())
-	return gobool(c)
-}
-
 // SetImage() is a wrapper around gtk_button_set_image().
 func (v *Button) SetImage(image IWidget) {
 	C.gtk_button_set_image(v.native(), image.toWidget())
@@ -5036,20 +5025,6 @@ func MenuNew() (*Menu, error) {
 	return wrapMenu(wrapObject(unsafe.Pointer(c))), nil
 }
 
-// PopupAtMouse() is a wrapper for gtk_menu_popup(), without the option for a custom positioning function.
-func (v *Menu) PopupAtMouseCursor(parentMenuShell IMenu, parentMenuItem IMenuItem, button int, activateTime uint32) {
-	wshell := nullableWidget(parentMenuShell)
-	witem := nullableWidget(parentMenuItem)
-
-	C.gtk_menu_popup(v.native(),
-		wshell,
-		witem,
-		nil,
-		nil,
-		C.guint(button),
-		C.guint32(activateTime))
-}
-
 // Popdown() is a wrapper around gtk_menu_popdown().
 func (v *Menu) Popdown() {
 	C.gtk_menu_popdown(v.native())
@@ -6967,15 +6942,6 @@ func (v *SizeGroup) SetMode(mode SizeGroupMode) {
 
 func (v *SizeGroup) GetMode() SizeGroupMode {
 	return SizeGroupMode(C.gtk_size_group_get_mode(v.native()))
-}
-
-func (v *SizeGroup) SetIgnoreHidden(ignoreHidden bool) {
-	C.gtk_size_group_set_ignore_hidden(v.native(), gbool(ignoreHidden))
-}
-
-func (v *SizeGroup) GetIgnoreHidden() bool {
-	c := C.gtk_size_group_get_ignore_hidden(v.native())
-	return gobool(c)
 }
 
 func (v *SizeGroup) AddWidget(widget IWidget) {

--- a/gtk/gtk_deprecated_since_3_20.go
+++ b/gtk/gtk_deprecated_since_3_20.go
@@ -1,0 +1,35 @@
+//+build gtk_3_6 gtk_3_8 gtk_3_10 gtk_3_12 gtk_3_14 gtk_3_16 gtk_3_18
+
+package gtk
+
+// #cgo pkg-config: gtk+-3.0
+// #include <gtk/gtk.h>
+// #include <stdlib.h>
+import "C"
+
+// GetFocusOnClick() is a wrapper around gtk_button_get_focus_on_click().
+func (v *Button) GetFocusOnClick() bool {
+	c := C.gtk_button_get_focus_on_click(v.native())
+	return gobool(c)
+}
+
+// BeginsTag is a wrapper around gtk_text_iter_begins_tag().
+func (v *TextIter) BeginsTag(v1 *TextTag) bool {
+	return gobool(C.gtk_text_iter_begins_tag(v.native(), v1.native()))
+}
+
+// ResizeToGeometry is a wrapper around gtk_window_resize_to_geometry().
+func (v *Window) ResizeToGeometry(width, height int) {
+	C.gtk_window_resize_to_geometry(v.native(), C.gint(width), C.gint(height))
+}
+
+// SetDefaultGeometry is a wrapper around gtk_window_set_default_geometry().
+func (v *Window) SetDefaultGeometry(width, height int) {
+	C.gtk_window_set_default_geometry(v.native(), C.gint(width),
+		C.gint(height))
+}
+
+// SetFocusOnClick() is a wrapper around gtk_button_set_focus_on_click().
+func (v *Button) SetFocusOnClick(focusOnClick bool) {
+	C.gtk_button_set_focus_on_click(v.native(), gbool(focusOnClick))
+}

--- a/gtk/gtk_deprecated_since_3_22.go
+++ b/gtk/gtk_deprecated_since_3_22.go
@@ -1,0 +1,44 @@
+//+build gtk_3_6 gtk_3_8 gtk_3_10 gtk_3_12 gtk_3_14 gtk_3_16 gtk_3_18 gtk_3_20
+
+package gtk
+
+// #cgo pkg-config: gtk+-3.0
+// #include <gtk/gtk.h>
+// #include <stdlib.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+// PopupAtMouse() is a wrapper for gtk_menu_popup(), without the option for a custom positioning function.
+func (v *Menu) PopupAtMouseCursor(parentMenuShell IMenu, parentMenuItem IMenuItem, button int, activateTime uint32) {
+	wshell := nullableWidget(parentMenuShell)
+	witem := nullableWidget(parentMenuItem)
+
+	C.gtk_menu_popup(v.native(),
+		wshell,
+		witem,
+		nil,
+		nil,
+		C.guint(button),
+		C.guint32(activateTime))
+}
+
+func (v *SizeGroup) GetIgnoreHidden() bool {
+	c := C.gtk_size_group_get_ignore_hidden(v.native())
+	return gobool(c)
+}
+
+// SetWMClass is a wrapper around gtk_window_set_wmclass().
+func (v *Window) SetWMClass(name, class string) {
+	cName := C.CString(name)
+	defer C.free(unsafe.Pointer(cName))
+	cClass := C.CString(class)
+	defer C.free(unsafe.Pointer(cClass))
+	C.gtk_window_set_wmclass(v.native(), (*C.gchar)(cName), (*C.gchar)(cClass))
+}
+
+func (v *SizeGroup) SetIgnoreHidden(ignoreHidden bool) {
+	C.gtk_size_group_set_ignore_hidden(v.native(), gbool(ignoreHidden))
+}

--- a/gtk/text_iter.go
+++ b/gtk/text_iter.go
@@ -96,11 +96,6 @@ func (v *TextIter) GetVisibleText(end *TextIter) string {
 	return C.GoString((*C.char)(c))
 }
 
-// BeginsTag is a wrapper around gtk_text_iter_begins_tag().
-func (v *TextIter) BeginsTag(v1 *TextTag) bool {
-	return gobool(C.gtk_text_iter_begins_tag(v.native(), v1.native()))
-}
-
 // EndsTag is a wrapper around gtk_text_iter_ends_tag().
 func (v *TextIter) EndsTag(v1 *TextTag) bool {
 	return gobool(C.gtk_text_iter_ends_tag(v.native(), v1.native()))

--- a/gtk/window.go
+++ b/gtk/window.go
@@ -105,12 +105,6 @@ func (v *Window) SetDefaultSize(width, height int) {
 	C.gtk_window_set_default_size(v.native(), C.gint(width), C.gint(height))
 }
 
-// SetDefaultGeometry is a wrapper around gtk_window_set_default_geometry().
-func (v *Window) SetDefaultGeometry(width, height int) {
-	C.gtk_window_set_default_geometry(v.native(), C.gint(width),
-		C.gint(height))
-}
-
 // GetScreen is a wrapper around gtk_window_get_screen().
 func (v *Window) GetScreen() (*gdk.Screen, error) {
 	c := C.gtk_window_get_screen(v.native())
@@ -328,15 +322,6 @@ func (v *Window) SetRole(s string) {
 	C.gtk_window_set_role(v.native(), cstr)
 }
 
-// SetWMClass is a wrapper around gtk_window_set_wmclass().
-func (v *Window) SetWMClass(name, class string) {
-	cName := C.CString(name)
-	defer C.free(unsafe.Pointer(cName))
-	cClass := C.CString(class)
-	defer C.free(unsafe.Pointer(cClass))
-	C.gtk_window_set_wmclass(v.native(), (*C.gchar)(cName), (*C.gchar)(cClass))
-}
-
 // GetDecorated is a wrapper around gtk_window_get_decorated().
 func (v *Window) GetDecorated() bool {
 	c := C.gtk_window_get_decorated(v.native())
@@ -490,11 +475,6 @@ func (v *Window) Move(x, y int) {
 // Resize is a wrapper around gtk_window_resize().
 func (v *Window) Resize(width, height int) {
 	C.gtk_window_resize(v.native(), C.gint(width), C.gint(height))
-}
-
-// ResizeToGeometry is a wrapper around gtk_window_resize_to_geometry().
-func (v *Window) ResizeToGeometry(width, height int) {
-	C.gtk_window_resize_to_geometry(v.native(), C.gint(width), C.gint(height))
 }
 
 // WindowSetDefaultIconFromFile is a wrapper around gtk_window_set_default_icon_from_file().

